### PR TITLE
osutil/disks: add DiskFromDevicePath, other misc changes

### PR DESCRIPTION
**Pull request by anonymouse64**
This is part of the multi-volume work and is the first of a couple small disks package PR's that will enable us to do all that must be done for multi-volume gadget asset updates on UC20.

### CODE REVIEW

File: osutil/disks/disks_linux.go [hunk 1]
```diff
@@ -145,6 +164,19 @@ var diskFromDeviceName = func(deviceName string) (Disk, error) {
 		major: major,
 		minor: minor,
 	}, nil
+
```
Comment (mardy) [inline osutil/disks/disks_linux.go:167]: nitpick: empty line

**Review by mardy — APPROVED** [review]
LGTM!

File: osutil/disks/disks_linux.go [hunk 2]
```diff
@@ -106,35 +117,43 @@ func parseUdevProperties(r io.Reader) (map[string]string, error) {
 	return m, scanner.Err()
 }
 
-// DiskFromDeviceName finds a matching Disk using the specified name, such as
```
Comment (mvo5) [inline osutil/disks/disks_linux.go:109]: Should the new `DiskFromDevicePath` also have a doc-string similar to the DiskFromDeviceName ?

File: osutil/disks/disks_linux.go [hunk 3]
```diff
@@ -145,6 +164,19 @@ var diskFromDeviceName = func(deviceName string) (Disk, error) {
 		major: major,
 		minor: minor,
 	}, nil
+
+}
+
+// diskFromDeviceName is exposed for mocking from other tests via
+// MockDeviceNameDisksToPartitionMapping.
+var diskFromDeviceName = func(deviceName string) (Disk, error) {
```
Comment (mvo5) [inline osutil/disks/disks_linux.go:172]: (nitpick^4) maybe move this close to `DiskFromDeviceName` ? 
[I was considering to use knuth arrow notation for this one btw ;)]

File: osutil/disks/mockdisk.go [hunk 4]
```diff
@@ -25,6 +25,10 @@ import (
 	"github.com/snapcore/snapd/osutil"
 )
 
+var (
```
Comment (mvo5) [inline osutil/disks/mockdisk.go:28]: same nitpick^N as above

**Review by mvo5 — APPROVED** [review]
Looks very nice, thank you! Some questions/nitpicks inline but no blockers - feel free to ignore.

File: osutil/disks/disks_linux.go (hunk 1 above)
Comment (anonymouse64) [reply osutil/disks/disks_linux.go:159]: removed

File: osutil/disks/disks_linux.go [hunk 5]
```diff
@@ -36,6 +36,10 @@ import (
 	"github.com/snapcore/snapd/osutil"
 )
 
+var (
```
[the remark this answers is not in the document]
Comment (anonymouse64) [reply osutil/disks/disks_linux.go]: done

File: osutil/disks/disks_linux.go (hunk 3 above)
Comment (anonymouse64) [reply osutil/disks/disks_linux.go:171]: done

File: osutil/disks/mockdisk.go (hunk 4 above)
Comment (anonymouse64) [reply osutil/disks/mockdisk.go]: done

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**